keps/3589-manage-jobs-selectively/README.md

```diff
@@ -0,0 +1,232 @@
+
+Currently, the primary mechanism for doing this is setting `manageJobsWithoutQueueName` to
+true.  When `manageJobsWithoutQueueName` is true, then Kueue will manage all instances of
+Kinds with Kueue integrations, even if they do not have a `kueue.x-k8s.io/queue-name` label.
+As a practical matter, Pods running in system namespaces must be exempted from being
+managed by Kueue even if `manageJobsWithoutQueueName` is true.  Therefore Kueue's Pod
+integration limits the scope of `manageJobsWithoutQueueName` by restricting its effect to
+Pods that are in namespaces that match against the `integrations.podOptions.namespaceSelector`.
+
+The recently added `Deployment` and `StatefulSet` integrations are built on top of the
+`Pod` integration and therefore the effect of `manageJobsWithoutQueueName` for these
+two integrations is also filtered by `integrations.podOptions.namespaceSelector`.  A similar
+rationale holds for `Deployment` and less strongly for `StatefulSet` that some mechanism
+to limit `manageJobsWithoutQueueName` is required otherwise it would be impossible in
+practice to ever set `manageJobsWithoutQueueName` to true.
+
+No other integrations beyond these three consider the `namespaceSelector` when deciding whether or not to
+suspend jobs that do not have a `queue-name` label.  This results in an irregular API and makes it
+harder to configure Kueue such that quotas will be enforced (cannot be bypassed by users).
+The irregularity between `batchv1/Job` and `Deployment` is likely to be especially confusing
+to users and cluster admins who would view both of these types as "built-in" to Kubernetes and
+would expect them to be configured in the same way.
+
+### Goals
+
+Provide a simple and uniform API to enable cluster admins to configure Kueue
+such that quotas will be enforced for specific namespaces.
+
+### Non-Goals
+
+Consider any scope for filtering `manageJobsWithoutQueueName` that is not based on namespaces.
+
+## Proposal
+
+We add a `managedJobsNamespaceSelector` of type `*metav1.LabelSelector`
+to the top level of the Kueue `Configuration` struct (same level as `manageJobsWithoutQueueName`).
+
+We use this new configuration for all integrations in the same way that the current
+`namespaceSelector` is used by the Pod integration.  Specifically,
+1. If `manageJobsWithoutQueueName` is false, `managedJobsNamespaceSelector` has no effect: Kueue will manage
+exactly those instances of supported Kinds that have a `queue-name` label.
+2. If `manageJobsWithoutQueueName` is true, then Kueue will (a) manage all instances of supported Kinds
+that have a `queue-name` label and (b) will manage all instances of supported Kinds that do not
+have a `queue-name` label if they are in namespaces that match `managedJobsNamespaceSelector`.
+
+We migrate away from using `podOptions.namespaceSelector` over the course of several releases.
+Ideally we remove it from the `Configuration` struct as part of going to the `v1` API level.
```

Comment:
Sgtm, I think we can say that in the transition period the actual namespace selector used for Pod integration (and thus StatefulSet and Deployments) will be the intersection of both `managedJobsNamespaceSelector` and `podOptions.namespaceSelector`.